docs/source/kernel-yarn-client-mode.md

```diff
@@ -0,0 +1,133 @@
+```
+
+* EG_REMOTE_HOSTS must be set to a comma-separated set of FQDN hosts indicating the hosts available for running kernels. (This can be specified via the command line as well: `--EnterpriseGatewayApp.remote_hosts`)
+
+```
+EG_REMOTE_HOSTS=elyra-node-1.fyre.ibm.com,elyra-node-2.fyre.ibm.com,elyra-node-3.fyre.ibm.com,elyra-node-4.fyre.ibm.com,elyra-node-5.fyre.ibm.com
+```
+
+For each supported Jupyter Kernel, we have provided sample kernel configurations and launchers as part of the release
+[e.g. jupyter_enterprise_gateway_kernelspecs-2.0.0.dev2.tar.gz](https://github.com/jupyter/enterprise_gateway/releases/download/v2.0.0rc1/jupyter_enterprise_gateway_kernelspecs-2.0.0rc1.tar.gz).
+
+Considering we would like to enable the IPython Kernel that comes pre-installed with Anaconda to run on
+Yarn Client mode, we would have to copy the sample configuration folder **spark_python_yarn_client**
+to where the Jupyter kernels are installed (e.g. jupyter kernelspec list)
+
+``` Bash
+wget https://github.com/jupyter/enterprise_gateway/releases/download/v2.0.0rc1/jupyter_enterprise_gateway_kernelspecs-2.0.0rc1.tar.gz
+SCALA_KERNEL_DIR="$(jupyter kernelspec list | grep -w "python3" | awk '{print $2}')"
+KERNELS_FOLDER="$(dirname "${SCALA_KERNEL_DIR}")"
+tar -zxvf enterprise_gateway_kernelspecs.tar.gz --strip 1 --directory $KERNELS_FOLDER/spark_python_yarn_client/ spark_python_yarn_client/
+```
+
+After that, you should have a kernel.json that looks similar to the one below:
+
+```json
+{
+  "language": "python",
+  "display_name": "Spark - Python (YARN Client Mode)",
+  "metadata": {
+    "process_proxy": {
+      "class_name": "enterprise_gateway.services.processproxies.distributed.DistributedProcessProxy"
+    }
+  },
+  "env": {
+    "SPARK_HOME": "/usr/hdp/current/spark2-client",
+    "PYSPARK_PYTHON": "/opt/conda/bin/python",
+    "PYTHONPATH": "${HOME}/.local/lib/python3.6/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.6-src.zip",
+    "SPARK_YARN_USER_ENV": "PYTHONUSERBASE=/home/yarn/.local,PYTHONPATH=${HOME}/.local/lib/python3.6/site-packages:/usr/hdp/current/spark2-client/python:/usr/hdp/current/spark2-client/python/lib/py4j-0.10.6-src.zip,PATH=/opt/conda/bin:$PATH",
+    "SPARK_OPTS": "--master yarn --deploy-mode client --name ${KERNEL_ID:-ERROR__NO__KERNEL_ID} --conf spark.yarn.submit.waitAppCompletion=false",
+    "LAUNCH_OPTS": ""
+  },
+  "argv": [
+    "/usr/local/share/jupyter/kernels/spark_python_yarn_client/bin/run.sh",
+     "--RemoteProcessProxy.kernel-id",
+    "{kernel_id}",
+    "--RemoteProcessProxy.response-address",
+    "{response_address}"
+  ]
+}
+```
+
+After making any necessary adjustments such as updating SPARK_HOME or other environment specific configuration, you now should have a new Kernel available which will use Jupyter Enterprise Gateway to execute your notebook cell contents.
+
+### Scala Kernel (Apache Toree kernel)
```

Comment:
The installation instructions appear duplicated in Yarn cluster and client but don't exist in standalone. 

I think we should probably have a section dedicated to kernel installation - at least for the non-containerized stuff.  Its important to point out that the kernels be installed on all nodes of the cluster with the exception of Apache Toree in Yarn cluster mode since spark-submit will transfer the files at startup.

For distributed modes (Yarn client and Spark Standalone and vanilla across a set of hosts) all kernels **and their corresponding kernelspecs** must reside on each node of the cluster because we essentially use a remote shell to invoke the kernel - so the kernelspec hierarchies (i.e., paths) must be identical.